### PR TITLE
Fix: Ignore subrequests metadata only in SiteSelector

### PR DESCRIPTION
**Pull request by BadPixxel**
## Subject
Site selector should not ignore Subrequests, but only metadata override.

If whole site use "host_with_path" or "host_with_path_by_locale", subrequests are rejected by router. 

I.e: in the admin: No route found for "GET /en/admin/en/core/get-short-object-description"

I am targeting this branch, because this is a bugfix for 3.X versions.

Related to  #1148.

## Changelog

```markdown
### Fixed
- Ignore metadata setup on SiteSelector, not whole site detection.
```

### CODE REVIEW

**Comment by BadPixxel** [comment]
Hi,

Any possibility to see this PR verified soon?

Thx!

**Comment by core23** [comment]
Can you provide a test for this?

**Comment by VincentLanglet** [comment]
Hi @BadPixxel, are you still interested in this PR ? Could you add a unit test for this ?

**Comment by BadPixxel** [comment]
Hi @VincentLanglet yes still need this PR. 
But no time to manage add unit tests up to now. I'll do my best to make it soon.

**Comment by BadPixxel** [comment]
Hi @VincentLanglet, just added few tests for this PR. Let me know if you see something I missed.

**Review by VincentLanglet — DISMISSED** [review]

File: tests/Site/SubRequestsSiteSelectorTest.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+use Sonata\PageBundle\Request\SiteRequest;
+use Sonata\PageBundle\Site\HostPathByLocaleSiteSelector;
+use Sonata\SeoBundle\Seo\SeoPageInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+/**
+ * @author BadPixxel <eshop.bpaquier@gmail.com>
+ */
+class SubRequestsSiteSelectorTest extends BaseLocaleSiteSelectorTest
```
Comment (core23) [inline tests/Site/SubRequestsSiteSelectorTest.php:27]: You can make this class final

**Review by core23 — DISMISSED** [review]

File: tests/Site/SubRequestsSiteSelectorTest.php (hunk 1 above)
Comment (BadPixxel) [reply tests/Site/SubRequestsSiteSelectorTest.php]: Done

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**